packages/immutadot/src/core/curry.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Manage a curried call.
+ * @returns {function|*} Either a new function or the result of the call to the actual function
+ * @this Object
+ */
+function curried(...newArgs) {
+  const { fn, last, minArity } = this
+
+  if (last) {
+    const [obj] = newArgs
+    return fn(obj, ...this.args)
+  }
+
+  const args = this.args.concat(newArgs)
```

Comment:
Why are you doing this on a separate line ?

Reply:
`args` is used 2 times in the destructuring